pkg/store/cas/cas.go

```diff
@@ -21,28 +21,37 @@ import (
 var ErrContentNotFound = errors.New("content not found")
 
 // CAS represents a content-addressable storage provider.
-// TODO (#344) Support writing and reading both v0 and v1 CIDs.
 type CAS struct {
-	cas       ariesstorage.Store
-	useV0CIDs bool
+	cas                ariesstorage.Store
+	useV0CIDsByDefault bool
 }
 
 // New returns a new CAS that uses the passed in provider as a backing store.
-func New(provider ariesstorage.Provider, useV0CIDs bool) (*CAS, error) {
+// useV0CIDsByDefault determines the default CID version to use when CAS.Write is called.
+func New(provider ariesstorage.Provider, useV0CIDsByDefault bool) (*CAS, error) {
 	cas, err := provider.OpenStore("cas_store")
 	if err != nil {
 		return nil, fmt.Errorf("failed to open store in underlying storage provider: %w", err)
 	}
 
-	return &CAS{cas: cas, useV0CIDs: useV0CIDs}, nil
+	return &CAS{cas: cas, useV0CIDsByDefault: useV0CIDsByDefault}, nil
 }
 
-// Write writes the given content to the underlying storage provider.
+// Write writes the given content to the underlying storage provider using this CAS' default CID version.
 // Returns the address of the content.
-// TODO (#418): Support creating IPFS-compatible CIDs when content is > 256KB.
 func (p *CAS) Write(content []byte) (string, error) {
+	return p.WriteWithCIDVersion(content, p.useV0CIDsByDefault)
+}
+
+// WriteWithCIDVersion writes the given content to the underlying storage provider.
+// If useV0CID is true, then the older v0 CID version will be used for calculating the address of the content instead
+// of the newer v1 version.
+// Returns the address of the content.
+// TODO (#418): Support creating IPFS-compatible CIDs when content is > 256KB.
+// TODO (#443): Support v1 CID formats (different multibases and multicodecs) other than just the IPFS default.
+func (p *CAS) WriteWithCIDVersion(content []byte, useV0CID bool) (string, error) {
```

Comment:
Instead of a boolean, I think it would be better to allow for flexible options that allow you to specify the CID version as well as potentially the multi-hash type, node type, etc. (e.g. https://github.com/trustbloc/fabric-peer-ext/blob/bfcd41db977e5b2dd7e9e5a7cf3dbf265849ba04/pkg/collections/offledger/dcas/client/dcasclient.go#L82). In this PR you need only have one option - WithCIDVersion, although this API is flexible enough to support other options if required.

Reply:
Done!